src/plugins/external_tool_support/src/blast/align_worker_subtasks/BlastAlignToReferenceTask.cpp

```diff
@@ -0,0 +1,371 @@
+    QString tempPath = AppContext::getAppSettings()->getUserAppsSettings()->getCurrentProcessTemporaryDirPath();
+    CHECK_EXT(!GUrlUtils::containSpaces(tempPath),
+              setError(tr("The task uses a temporary folder to process the data. The folder path is required not to have spaces. "
+                          "Please set up an appropriate path for the \"Temporary files\" parameter on the \"Directories\" tab of the UGENE Application Settings.")), );
+
+    U2Region readsRange = readsRangePerSubtask.takeFirst();
+    addSubTask(createNewSubtask(readsRange));
+}
+
+QList<Task *> BlastAlignToReferenceMuxTask::onSubTaskFinished(Task *) {
+    CHECK_OP(stateInfo, {});
+    CHECK(!readsRangePerSubtask.isEmpty(), {});
+    U2Region readsRange = readsRangePerSubtask.takeFirst();
+    return {createNewSubtask(readsRange)};
+}
+
+BlastAlignToReferenceTask *BlastAlignToReferenceMuxTask::createNewSubtask(const U2Region &readsRange) const {
+    QList<SharedDbiDataHandler> readsPerSubtask = reads.mid(readsRange.startPos, readsRange.length);
+    QString subtaskNameSuffix = tr(", reads range %1-%2").arg(readsRange.startPos).arg(readsRange.endPos());
+    return new BlastAlignToReferenceTask(blastDbPath, readsPerSubtask, reference, storage, subtaskNameSuffix);
+}
+
+Task::ReportResult BlastAlignToReferenceMuxTask::report() {
+    QList<QPointer<Task>> subtasks = getSubtasks();
+    for (const auto &task : qAsConst(subtasks)) {
+        auto subtask = qobject_cast<BlastAlignToReferenceTask *>(task.data());
+        SAFE_POINT(subtask != nullptr, "BlastAlignToReferenceMuxTask has wrong subtask", ReportResult_Finished);
+        alignmentResults << subtask->getAlignmentResults();
+    }
+    return ReportResult_Finished;
+}
+
+const QList<AlignToReferenceResult> &BlastAlignToReferenceMuxTask::getAlignmentResults() const {
+    return alignmentResults;
+}
+
+/************************************************************************/
+/* BlastAlignToReferenceTask */
+/************************************************************************/
+BlastAlignToReferenceTask::BlastAlignToReferenceTask(const QString &_blastDbPath,
+                                                     const QList<SharedDbiDataHandler> &_reads,
+                                                     const SharedDbiDataHandler &_reference,
+                                                     DbiDataStorage *_storage,
+                                                     const QString &taskNameSuffix)
+    : Task(tr("Align reads with BLAST & Smith-Waterman task") + taskNameSuffix, TaskFlags_NR_FOSE_COSC),
+      dbPath(_blastDbPath),
+      reads(_reads),
+      reference(_reference),
+      storage(_storage) {
+    QScopedPointer<U2SequenceObject> referenceSequence(StorageUtils::getSequenceObject(storage, reference));
+    CHECK_EXT(referenceSequence->getSequenceLength() < INT_MAX, setError(tr("Maximum supported reference sequence length is 2Gb")), );
```

Comment:
`std::numeric_limits<int>::max()` more modern

Reply:
I prefer the old way as more readable: there is nothing wrong with constants